.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,25 @@
+---
+    name: pre-commit
+
+    on:
+      pull_request:
+
+    jobs:
+      lint-pr-title:
+        name: Validate PR titles
+        runs-on: ubuntu-latest
+        steps:
+        - uses: amannn/action-semantic-pull-request@v5
```

Comment:
This does not work as-is for a public repository:

https://github.com/amannn/action-semantic-pull-request#event-triggers

Anything that needs to use a `GITHUB_TOKEN` must use the `pull_request_target` trigger. And that should be introduced separately, as `pull_request_target` runs on the default branch of the target repository, meaning the initial pull request to add the workflow won't actually run the workflow. Only after it's merged will subsequent pull requests trigger it.